### ISSUE TITLE
Error when invalid URL in href (ValueError: Invalid IPv6 URL)

### ISSUE THREAD

**Issue by antoineleclair** [mozilla/bleach#376]
Using `bleach==2.1.3` and Python 3.6.5:

```python
Python 3.6.5 (default, Jun  6 2018, 19:19:24) 
[GCC 6.3.0 20170516] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import bleach
>>> bleach.clean('<a href="http://example.com]">boom</a>')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/local/lib/python3.6/site-packages/bleach/__init__.py", line 99, in clean
    return cleaner.clean(text)
  File "/usr/local/lib/python3.6/site-packages/bleach/sanitizer.py", line 282, in clean
    return self.serializer.render(filtered)
  File "/usr/local/lib/python3.6/site-packages/html5lib/serializer.py", line 398, in render
    return "".join(list(self.serialize(treewalker)))
  File "/usr/local/lib/python3.6/site-packages/bleach/sanitizer.py", line 751, in serialize
    for stoken in super(BleachHTMLSerializer, self).serialize(treewalker, encoding):
  File "/usr/local/lib/python3.6/site-packages/html5lib/serializer.py", line 265, in serialize
    for token in treewalker:
  File "/usr/local/lib/python3.6/site-packages/bleach/sanitizer.py", line 439, in __iter__
    ret = self.sanitize_token(token)
  File "/usr/local/lib/python3.6/site-packages/bleach/sanitizer.py", line 469, in sanitize_token
    return self.allow_token(token)
  File "/usr/local/lib/python3.6/site-packages/bleach/sanitizer.py", line 620, in allow_token
    new_value = self.sanitize_uri_value(val, self.allowed_protocols)
  File "/usr/local/lib/python3.6/site-packages/bleach/sanitizer.py", line 575, in sanitize_uri_value
    parsed = urlparse(new_value)
  File "/usr/local/lib/python3.6/urllib/parse.py", line 368, in urlparse
    splitresult = urlsplit(url, scheme, allow_fragments)
  File "/usr/local/lib/python3.6/urllib/parse.py", line 418, in urlsplit
    raise ValueError("Invalid IPv6 URL")
ValueError: Invalid IPv6 URL
```

This works using `bleach==1.5.0` (that's the version we used before upgrading to 2.1.3).

**Comment by willkg** [comment]
What version of html5lib are you using and what are you expecting that string to look like after it's cleaned?

**Comment by antoineleclair** [comment]
`bleach==2.1.3` with `html5lib==1.0.1`.

With `bleach==1.5.0` and `html5lib==0.9999999`, the URL is kept unmodified. I think that's what I would expect. For sure, whatever happens to the URL, I would not want an exception raised and would like to see the rest of the markup cleaned.

**Comment by willkg** [comment]
That makes sense!

Mmm... "Invalid IPv6 URL" is not an error I've ever seen before, so I don't have any ideas about what the crux of the problem is or what we can do to fix it.

I also don't have time to look into this for a long while. Do you have time to look into this?

**Comment by antoineleclair** [comment]
Looks like `sanitize_uri_value` https://github.com/mozilla/bleach/blob/e1a6ba29495188eedca14b6a3884cb4e94732056/bleach/sanitizer.py#L549 behavior is to return `None` if the URI is invalid.

I could make a pull request that does that. I would add a `try`/`except` around parsing the URI https://github.com/mozilla/bleach/blob/e1a6ba29495188eedca14b6a3884cb4e94732056/bleach/sanitizer.py#L581 and if a `ValueError` is raised, it would return `None`.

Would that make sense?

**Comment by willkg** [comment]
I don't have time to look at it, but nothing you're suggesting jumps out at me as a bad idea.

What's `new_value` at that point with the url that's causing problems?

**Comment by antoineleclair** [comment]
`new_value` is the string `http://example.com]` at that point.

Let me know if that pull request :point_up: makes sense or if I should do any change to it.

**Comment by willkg** [comment]
This was fixed in #378 which I just merged. It'll go out in the next release--probably 2.1.4.

**Comment by willkg** [comment]
@antoineleclair Thank you for working on this!
